lib/task/import/validate/csvValidatorCollection.php

```diff
@@ -80,39 +98,21 @@ public function getResultCollection(): CsvValidatorResultCollection
      * the correct Validator objects for the specified $className.
      *
      * @param string $className             The entity name. e.g. QubitInformationObject
-     * @param array  $options               the options array passed to the validator
-     * @param string $specificValidatorList comma separated string of validator class names
+     * @param array  $options               The options array passed to the validator
+     * @param string $specificValidatorList Comma separated string of validator class names
      */
     public static function getValidatorCollection(string $className, array &$options, ?string $validatorList = null): CsvValidatorCollection
     {
-        // Map of available validator classes.
-        $validatorsAvailable = [
-            'CsvSampleValuesValidator' => CsvSampleValuesValidator::class,
-            'CsvFileEncodingValidator' => CsvFileEncodingValidator::class,
-            'CsvColumnNameValidator' => CsvColumnNameValidator::class,
-            'CsvColumnCountValidator' => CsvColumnCountValidator::class,
-            'CsvDuplicateColumnNameValidator' => CsvDuplicateColumnNameValidator::class,
-            'CsvEmptyRowValidator' => CsvEmptyRowValidator::class,
-            'CsvCultureValidator' => CsvCultureValidator::class,
-            'CsvLanguageValidator' => CsvLanguageValidator::class,
-            'CsvFieldLengthValidator' => CsvFieldLengthValidator::class,
-            'CsvParentValidator' => CsvParentValidator::class,
-            'CsvLegacyIdValidator' => CsvLegacyIdValidator::class,
-            'CsvEventValuesValidator' => CsvEventValuesValidator::class,
-            'CsvDigitalObjectPathValidator' => CsvDigitalObjectPathValidator::class,
-            'CsvDigitalObjectUriValidator' => CsvDigitalObjectUriValidator::class,
-        ];
-
         // Remove tests that are not applicable to this $className.
-        foreach ($validatorsAvailable as $key => $class) {
-            if (!in_array($className, $class::LIMIT_TO) && !empty($class::LIMIT_TO)) {
-                unset($validatorsAvailable[$key]);
+        foreach (self::$validatorsAvailable as $key => $class) {
+            if (!empty($class::LIMIT_TO) && !in_array($className, $class::LIMIT_TO)) {
+                unset(self::$validatorsAvailable[$key]);
```

Comment:
Because `self::$validatorsAvailable` is a static property, I think this will remove the validator from $validatorsAvailable for ALL instances of `CsvValidatorCollection`.  I think you'll want to create a local working copy of `self::$validatorsAvailable`.

Reply:
Thanks @djjuhasz - Added commit that uses a working copy of the validator list. Also added a couple basic PHPUnit tests for the CsvValidatorCollection class.